<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Using mutmatch for selection estimation</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Using mutmatch for selection
estimation</h1>



<p>This vignette demonstrates how to use the <em>mutmatch</em> package
to estimate selection pressures and epistasis in somatic cells by
utilizing data from neighboring genes to establish a baseline mutation
rate..</p>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>The <strong>mutmatch</strong> package provides tools for analyzing
mutation data to estimate selection pressures in cancer genes. In this
vignette, we will demonstrate the package’s capabilities using example
datasets that are included for ease of learning. These datasets feature
real mutation data and gene-level copy number alterations from 60 TCGA
samples spanning three cancer types: COREAD, LUAD, and SKCM. All
provided data is accessible openly, allowing for transparent and
reproducible analysis.</p>
</div>
<div id="loading-the-package" class="section level2">
<h2>Loading the Package</h2>
<p>Load the <em>mutmatch</em> package into your R session:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(mutmatch)</span></code></pre></div>
</div>
<div id="basic-usage" class="section level2">
<h2>Basic usage</h2>
<p>Our primary focus is to demonstrate how to estimate selection
pressures acting on specific genes, such as the <em>KRAS</em> gene, by
using the provided sample data. Below is a command designed to estimate
selection on <em>KRAS</em> when in a diploid state, as well as to assess
the change in selection pressure resulting from the loss of one gene
copy:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>selection_estimates <span class="ot">&lt;-</span> <span class="fu">get_selection_estimates_neighbors</span>(</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>  <span class="at">hgnc =</span> <span class="st">&quot;KRAS&quot;</span>,</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>  <span class="at">mutationsPath =</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;example_mutations.csv.gz&quot;</span>, <span class="at">package =</span> <span class="st">&quot;mutmatch&quot;</span>),</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>  <span class="at">annotationGenePath =</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;example_gene_annotation.csv.gz&quot;</span>, <span class="at">package =</span> <span class="st">&quot;mutmatch&quot;</span>),</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>  <span class="at">annotationGenomeWidePath =</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;example_genomewide_annotation.csv&quot;</span>, <span class="at">package =</span> <span class="st">&quot;mutmatch&quot;</span>),</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>  <span class="at">neighborsWindow =</span> <span class="st">&quot;0.5Mb&quot;</span>,</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>  <span class="at">outlierNeighborsThreshold =</span> <span class="fl">0.2</span>,</span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a>  <span class="at">formula =</span> <span class="st">&quot;MutationNumber ~ isTarget + CNA + isTarget:CNA + Mutation + offset(log(ntAtRisk))&quot;</span>,</span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>  <span class="at">family =</span> <span class="st">&quot;bayes.poisson&quot;</span>,</span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a>  <span class="at">simtimes =</span> <span class="dv">50</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a>)</span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="co">#&gt; Retrieving mutation counts using neighbors baseline mutation model</span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="co">#&gt; Fitting the selection model and debiasing estimates</span></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a><span class="co">#&gt; Simulating data using neutral selection model</span></span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a><span class="co">#&gt; Performing model fitting on real data for each cohort</span></span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a><span class="co">#&gt; Performing model fitting on simulated data for each cohort</span></span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a><span class="co">#&gt; Correcting selection estimates using simulated data</span></span></code></pre></div>
<div id="explanation-of-variables-in-the-model" class="section level3">
<h3>Explanation of Variables in the Model:</h3>
<ul>
<li><code>isTarget</code> (<span class="math inline">\(t\)</span>): A
binary indicator that determines whether mutations occur in the gene of
interest versus background genes. It serves as a proxy for
selection.</li>
<li><code>CNA</code> (<span class="math inline">\(c\)</span>): Adjusts
for the impact of copy number alterations on mutation rates.</li>
<li><code>isTarget:CNA</code> (<span class="math inline">\(tc\)</span>):
Evaluates conditional selection pressures in samples with varying copy
number alterations.</li>
<li><code>Mutation</code> (<span class="math inline">\(m_{i}\)</span>):
Controls for biases introduced by different mutation spectra, where
<span class="math inline">\(i\)</span> denotes different mutation
types.</li>
<li><code>ntAtRisk</code> (<span class="math inline">\(r\)</span>):
Scales estimates across DNA regions of varying lengths.</li>
<li><code>Intercept</code> (<span class="math inline">\(\alpha\)</span>): Represents the baseline level of
mutation rate in the absence of other factors, serving as a reference
point for the model.</li>
</ul>
<p>The formal model expression, using a logarithmic link function, is
presented as follows: <span class="math display">\[ \log E[Y] = \omega t
+ \gamma c + \delta tc + \sum_{i} \mu_i m_i + \alpha + \log r
\]</span></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">head</span>(selection_estimates)</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="co">#&gt; Key: &lt;Cohort&gt;</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="co">#&gt;    Cohort        coefName         coef coef_corrected         pVal</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="co">#&gt;    &lt;char&gt;          &lt;char&gt;        &lt;num&gt;          &lt;num&gt;        &lt;num&gt;</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="co">#&gt; 1: COREAD       Intercept -12.69255476   -12.69255476 3.117515e-17</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="co">#&gt; 2: COREAD       isTarget1   3.27819938     3.56423363 1.191858e-02</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a><span class="co">#&gt; 3: COREAD           CNA-1   0.00000000     0.00000000 1.000000e+00</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a><span class="co">#&gt; 4: COREAD isTarget1:CNA-1   0.00000000     0.00000000 1.000000e+00</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a><span class="co">#&gt; 5: COREAD   MutationACA&gt;G  -0.01132096    -0.01132096 9.908600e-01</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a><span class="co">#&gt; 6: COREAD   MutationACA&gt;T  -0.01132096    -0.01132096 9.908600e-01</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a><span class="co">#&gt;    upperTailPval lowerTailPval Std.error converged coef_sim_median</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a><span class="co">#&gt;            &lt;num&gt;         &lt;num&gt;     &lt;num&gt;    &lt;lgcl&gt;           &lt;num&gt;</span></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a><span class="co">#&gt; 1:   1.000000000     0.0000000  1.503487      TRUE   -10.982641672</span></span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a><span class="co">#&gt; 2:   0.005959291     0.9940407  1.303694      TRUE    -0.286034244</span></span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a><span class="co">#&gt; 3:   0.500000000     0.5000000  1.000000      TRUE     0.000000000</span></span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a><span class="co">#&gt; 4:   0.500000000     0.5000000  1.000000      TRUE     0.000000000</span></span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a><span class="co">#&gt; 5:   0.504570011     0.4954300  0.988249      TRUE    -0.006500099</span></span>
<span id="cb3-18"><a href="#cb3-18" tabindex="-1"></a><span class="co">#&gt; 6:   0.504570011     0.4954300  0.988249      TRUE    -0.006500099</span></span>
<span id="cb3-19"><a href="#cb3-19" tabindex="-1"></a><span class="co">#&gt;                                                                                                                                                                                                                                                                                                                                                                                                           coef_sim</span></span>
<span id="cb3-20"><a href="#cb3-20" tabindex="-1"></a><span class="co">#&gt;                                                                                                                                                                                                                                                                                                                                                                                                             &lt;char&gt;</span></span>
<span id="cb3-21"><a href="#cb3-21" tabindex="-1"></a><span class="co">#&gt; 1: -11.096 -11.096 -10.983 -11.096 -10.983 -10.983 -11.096 -10.983 -10.983 -10.983 -10.983 -11.096 -11.096 -10.983 -11.096 -11.096 -11.431 -10.983 -10.983 -10.983 -11.096 -10.983 -11.431 -11.096 -10.983 -10.983 -10.983 -11.096 -10.983 -11.096 -11.096 -11.096 -10.983 -10.983 -10.983 -11.096 -10.983 -10.983 -11.096 -10.983 -10.983 -10.983 -10.983 -11.431 -11.096 -11.096 -11.096 -10.983 -11.096 -10.983</span></span>
<span id="cb3-22"><a href="#cb3-22" tabindex="-1"></a><span class="co">#&gt; 2:                                                                          0.354 0.354 -0.286 0.354 -0.286 -0.286 0.354 -0.286 -0.286 -0.286 -0.286 0.354 0.354 -0.286 0.354 0.354 1.376 -0.286 -0.286 -0.286 0.354 -0.286 1.376 0.354 -0.286 -0.286 -0.286 0.354 -0.286 0.354 0.354 0.354 -0.286 -0.286 -0.286 0.354 -0.286 -0.286 0.354 -0.286 -0.286 -0.286 -0.286 1.376 0.354 0.354 0.354 -0.286 0.354 -0.286</span></span>
<span id="cb3-23"><a href="#cb3-23" tabindex="-1"></a><span class="co">#&gt; 3:                                                                                                                                                                                                                                                                                                             0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0</span></span>
<span id="cb3-24"><a href="#cb3-24" tabindex="-1"></a><span class="co">#&gt; 4:                                                                                                                                                                                                                                                                                                             0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0</span></span>
<span id="cb3-25"><a href="#cb3-25" tabindex="-1"></a><span class="co">#&gt; 5:                                                   -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.009 -0.007 -0.007 -0.007 -0.007 -0.007 -0.009 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.009 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007</span></span>
<span id="cb3-26"><a href="#cb3-26" tabindex="-1"></a><span class="co">#&gt; 6:                                                   -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.009 -0.007 -0.007 -0.007 -0.007 -0.007 -0.009 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007 -0.009 -0.007 -0.007 -0.007 -0.007 -0.007 -0.007</span></span>
<span id="cb3-27"><a href="#cb3-27" tabindex="-1"></a><span class="co">#&gt;                                                                                                                                                                                                                                                                                                                                                pVal_neglog10_sim</span></span>
<span id="cb3-28"><a href="#cb3-28" tabindex="-1"></a><span class="co">#&gt;                                                                                                                                                                                                                                                                                                                                                           &lt;char&gt;</span></span>
<span id="cb3-29"><a href="#cb3-29" tabindex="-1"></a><span class="co">#&gt; 1: 25.873 25.873 25.882 25.873 25.882 25.882 25.873 25.882 25.882 25.882 25.882 25.873 25.873 25.882 25.873 25.873 24.647 25.882 25.882 25.882 25.873 25.882 24.647 25.873 25.882 25.882 25.882 25.873 25.882 25.873 25.873 25.873 25.882 25.882 25.882 25.873 25.882 25.882 25.873 25.882 25.882 25.882 25.882 24.647 25.873 25.873 25.873 25.882 25.873 25.882</span></span>
<span id="cb3-30"><a href="#cb3-30" tabindex="-1"></a><span class="co">#&gt; 2:                                                   0.188 0.188 0.137 0.188 0.137 0.137 0.188 0.137 0.137 0.137 0.137 0.188 0.188 0.137 0.188 0.188 0.932 0.137 0.137 0.137 0.188 0.137 0.932 0.188 0.137 0.137 0.137 0.188 0.137 0.188 0.188 0.188 0.137 0.137 0.137 0.188 0.137 0.137 0.188 0.137 0.137 0.137 0.137 0.932 0.188 0.188 0.188 0.137 0.188 0.137</span></span>
<span id="cb3-31"><a href="#cb3-31" tabindex="-1"></a><span class="co">#&gt; 3:                                                                                                                                                                                                                                                           0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0</span></span>
<span id="cb3-32"><a href="#cb3-32" tabindex="-1"></a><span class="co">#&gt; 4:                                                                                                                                                                                                                                                           0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0</span></span>
<span id="cb3-33"><a href="#cb3-33" tabindex="-1"></a><span class="co">#&gt; 5:                                                   0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.003 0.002 0.002 0.002 0.002 0.002 0.003 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.003 0.002 0.002 0.002 0.002 0.002 0.002</span></span>
<span id="cb3-34"><a href="#cb3-34" tabindex="-1"></a><span class="co">#&gt; 6:                                                   0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.003 0.002 0.002 0.002 0.002 0.002 0.003 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.002 0.003 0.002 0.002 0.002 0.002 0.002 0.002</span></span>
<span id="cb3-35"><a href="#cb3-35" tabindex="-1"></a><span class="co">#&gt;    MutN_isTarget_control_CNA_control MutN_isTarget_control_CNA_test</span></span>
<span id="cb3-36"><a href="#cb3-36" tabindex="-1"></a><span class="co">#&gt;                                &lt;int&gt;                          &lt;int&gt;</span></span>
<span id="cb3-37"><a href="#cb3-37" tabindex="-1"></a><span class="co">#&gt; 1:                                 0                              0</span></span>
<span id="cb3-38"><a href="#cb3-38" tabindex="-1"></a><span class="co">#&gt; 2:                                 0                              0</span></span>
<span id="cb3-39"><a href="#cb3-39" tabindex="-1"></a><span class="co">#&gt; 3:                                 0                              0</span></span>
<span id="cb3-40"><a href="#cb3-40" tabindex="-1"></a><span class="co">#&gt; 4:                                 0                              0</span></span>
<span id="cb3-41"><a href="#cb3-41" tabindex="-1"></a><span class="co">#&gt; 5:                                 0                              0</span></span>
<span id="cb3-42"><a href="#cb3-42" tabindex="-1"></a><span class="co">#&gt; 6:                                 0                              0</span></span>
<span id="cb3-43"><a href="#cb3-43" tabindex="-1"></a><span class="co">#&gt;    MutN_isTarget_test_CNA_control MutN_isTarget_test_CNA_test</span></span>
<span id="cb3-44"><a href="#cb3-44" tabindex="-1"></a><span class="co">#&gt;                             &lt;int&gt;                       &lt;int&gt;</span></span>
<span id="cb3-45"><a href="#cb3-45" tabindex="-1"></a><span class="co">#&gt; 1:                              3                           0</span></span>
<span id="cb3-46"><a href="#cb3-46" tabindex="-1"></a><span class="co">#&gt; 2:                              3                           0</span></span>
<span id="cb3-47"><a href="#cb3-47" tabindex="-1"></a><span class="co">#&gt; 3:                              3                           0</span></span>
<span id="cb3-48"><a href="#cb3-48" tabindex="-1"></a><span class="co">#&gt; 4:                              3                           0</span></span>
<span id="cb3-49"><a href="#cb3-49" tabindex="-1"></a><span class="co">#&gt; 5:                              3                           0</span></span>
<span id="cb3-50"><a href="#cb3-50" tabindex="-1"></a><span class="co">#&gt; 6:                              3                           0</span></span>
<span id="cb3-51"><a href="#cb3-51" tabindex="-1"></a><span class="co">#&gt;    NtN_isTarget_control_CNA_control NtN_isTarget_control_CNA_test</span></span>
<span id="cb3-52"><a href="#cb3-52" tabindex="-1"></a><span class="co">#&gt;                               &lt;num&gt;                         &lt;num&gt;</span></span>
<span id="cb3-53"><a href="#cb3-53" tabindex="-1"></a><span class="co">#&gt; 1:                            14400                             0</span></span>
<span id="cb3-54"><a href="#cb3-54" tabindex="-1"></a><span class="co">#&gt; 2:                            14400                             0</span></span>
<span id="cb3-55"><a href="#cb3-55" tabindex="-1"></a><span class="co">#&gt; 3:                            14400                             0</span></span>
<span id="cb3-56"><a href="#cb3-56" tabindex="-1"></a><span class="co">#&gt; 4:                            14400                             0</span></span>
<span id="cb3-57"><a href="#cb3-57" tabindex="-1"></a><span class="co">#&gt; 5:                            14400                             0</span></span>
<span id="cb3-58"><a href="#cb3-58" tabindex="-1"></a><span class="co">#&gt; 6:                            14400                             0</span></span>
<span id="cb3-59"><a href="#cb3-59" tabindex="-1"></a><span class="co">#&gt;    NtN_isTarget_test_CNA_control NtN_isTarget_test_CNA_test</span></span>
<span id="cb3-60"><a href="#cb3-60" tabindex="-1"></a><span class="co">#&gt;                            &lt;num&gt;                      &lt;num&gt;</span></span>
<span id="cb3-61"><a href="#cb3-61" tabindex="-1"></a><span class="co">#&gt; 1:                          2775                          0</span></span>
<span id="cb3-62"><a href="#cb3-62" tabindex="-1"></a><span class="co">#&gt; 2:                          2775                          0</span></span>
<span id="cb3-63"><a href="#cb3-63" tabindex="-1"></a><span class="co">#&gt; 3:                          2775                          0</span></span>
<span id="cb3-64"><a href="#cb3-64" tabindex="-1"></a><span class="co">#&gt; 4:                          2775                          0</span></span>
<span id="cb3-65"><a href="#cb3-65" tabindex="-1"></a><span class="co">#&gt; 5:                          2775                          0</span></span>
<span id="cb3-66"><a href="#cb3-66" tabindex="-1"></a><span class="co">#&gt; 6:                          2775                          0</span></span></code></pre></div>
<p>Additionally, along with the selection estimates, raw mutation counts
and the number of nucleotides-at-risk are detailed for each region under
various states:</p>
<ul>
<li><code>*_isTarget_control_CNA_control</code>: Represents neighboring
genes in the baseline (diploid) CNA state.</li>
<li><code>*_isTarget_control_CNA_test</code>: Represents neighboring
genes in the tested CNA state (deleted in this context).</li>
<li><code>*_isTarget_test_CNA_control</code>: Refers to the central gene
(<em>KRAS</em>) in its diploid CNA state.</li>
<li><code>*_isTarget_test_CNA_test</code>: Refers to the <em>KRAS</em>
gene in samples where a deletion has occurred.</li>
</ul>
<p>The prefix <em>MutN</em> denotes mutation counts, while <em>NtN</em>
refers to the number of nucleotides-at-risk.</p>
</div>
<div id="postprocessing" class="section level3">
<h3>Postprocessing</h3>
<p>Postprocessing is essential for refining initial estimates and
addressing potential biases. For a detailed explanation of this process,
please consult the <a href="https://github.com/ebesedina/mutmatch#postprocessing">Postprocessing
section</a> or the related publication. The
<code>get_selection_estimates_neighbors</code> function’s output
includes the columns <code>coef_sim</code> and
<code>pVal_neglog10_sim</code>, which contain the distributions of
selection estimates and their corresponding p-values based on a
simulated (randomized) dataset. This data helps correct for
distributional biases:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>selection_estimates <span class="ot">=</span> <span class="fu">post_process_pvalues</span>(selection_estimates)</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>selection_estimates[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="fu">c</span>(<span class="st">&quot;Cohort&quot;</span>, <span class="st">&quot;coefName&quot;</span>, <span class="st">&quot;coef&quot;</span>, <span class="st">&quot;coef_corrected&quot;</span>, <span class="st">&quot;pVal&quot;</span>,<span class="st">&quot;pVal_corrected&quot;</span>)]</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="co">#&gt; Key: &lt;Cohort&gt;</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="co">#&gt;    Cohort        coefName         coef coef_corrected         pVal</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="co">#&gt;    &lt;char&gt;          &lt;char&gt;        &lt;num&gt;          &lt;num&gt;        &lt;num&gt;</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="co">#&gt; 1: COREAD       Intercept -12.69255476   -12.69255476 3.117515e-17</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="co">#&gt; 2: COREAD       isTarget1   3.27819938     3.56423363 1.191858e-02</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a><span class="co">#&gt; 3: COREAD           CNA-1   0.00000000     0.00000000 1.000000e+00</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a><span class="co">#&gt; 4: COREAD isTarget1:CNA-1   0.00000000     0.00000000 1.000000e+00</span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a><span class="co">#&gt; 5: COREAD   MutationACA&gt;G  -0.01132096    -0.01132096 9.908600e-01</span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a><span class="co">#&gt;    pVal_corrected</span></span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a><span class="co">#&gt;             &lt;num&gt;</span></span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a><span class="co">#&gt; 1:             NA</span></span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a><span class="co">#&gt; 2:              0</span></span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a><span class="co">#&gt; 3:             NA</span></span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a><span class="co">#&gt; 4:              1</span></span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a><span class="co">#&gt; 5:             NA</span></span></code></pre></div>
</div>
</div>
<div id="inputs-and-default-parameters" class="section level2">
<h2>Inputs and Default Parameters</h2>
<p>In this section, we delve into the details of the input data formats
and the default parameters used in the
<code>get_selection_estimates_neighbors</code> function. Understanding
these inputs is crucial for effective use of the <em>mutmatch</em>
package.</p>
<div id="mutation-data" class="section level3">
<h3>Mutation Data</h3>
<p>The <code>mutationPath</code> parameter requires a path to the CSV
file containing detailed mutation positions. This file must include
several columns:</p>
<ul>
<li><code>CHROM</code>: Chromosome number.</li>
<li><code>POS</code>: Mutation position on the chromosome.</li>
<li><code>REF</code>: Reference nucleotide in normal cells.</li>
<li><code>ALT</code>: Altered nucleotide representing the mutation.</li>
<li><code>Sample</code>: Identifier for the sample in which the mutation
was observed.</li>
</ul>
<p>It’s important to note that the mutation data should correspond to
the GRCh37/hg19 reference genome assembly to ensure compatibility with
the analysis framework.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>mutationData <span class="ot">=</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, </span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="st">&quot;example_mutations.csv.gz&quot;</span>, <span class="at">package =</span> <span class="st">&quot;mutmatch&quot;</span>))</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="fu">head</span>(mutationData)</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="co">#&gt;     CHROM       POS    REF    ALT                       Sample</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="co">#&gt;    &lt;char&gt;     &lt;int&gt; &lt;char&gt; &lt;char&gt;                       &lt;char&gt;</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="co">#&gt; 1:  chr17  40997947      G      A TCGA-AA-A01X-01A-21W-A096-10</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="co">#&gt; 2:  chr12  25398284      C      A TCGA-AA-A01X-01A-21W-A096-10</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a><span class="co">#&gt; 3:   chr9 133954657      C      T TCGA-AA-A01X-01A-21W-A096-10</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a><span class="co">#&gt; 4:  chr19  43259170      G      A TCGA-AA-A01X-01A-21W-A096-10</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a><span class="co">#&gt; 5:  chr21  45502758      G      T TCGA-AA-A01X-01A-21W-A096-10</span></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a><span class="co">#&gt; 6:   chr8   1575018      G      A TCGA-AA-A01X-01A-21W-A096-10</span></span></code></pre></div>
</div>
<div id="genomic-annotation" class="section level3">
<h3>Genomic Annotation</h3>
<p>Each sample in the dataset should be accompanied by relevant genomic
annotations, which in this example includes the type of cancer:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>genomewide_annotation <span class="ot">=</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, </span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="st">&quot;example_genomewide_annotation.csv&quot;</span>, <span class="at">package =</span> <span class="st">&quot;mutmatch&quot;</span>))</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="fu">head</span>(genomewide_annotation)</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="co">#&gt;    Cancer                       Sample</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="co">#&gt;    &lt;char&gt;                       &lt;char&gt;</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="co">#&gt; 1:   LUAD TCGA-44-2655-01A-01D-1553-08</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="co">#&gt; 2:   LUAD TCGA-MP-A4TF-01A-11D-A25L-08</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="co">#&gt; 3:   LUAD TCGA-75-7027-01A-11D-1945-08</span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a><span class="co">#&gt; 4:   LUAD TCGA-50-8457-01A-11D-2323-08</span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a><span class="co">#&gt; 5:   LUAD TCGA-44-6776-01A-11D-1855-08</span></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a><span class="co">#&gt; 6:   LUAD TCGA-55-8512-01A-11D-2393-08</span></span></code></pre></div>
</div>
<div id="copy-number-alterations-cna-data" class="section level3">
<h3>Copy Number Alterations (CNA) Data</h3>
<p>CNA data provides context about gene dosage variations across
different samples. This dataset indicates whether there is a normal
diploid state (CNA = 0) or a gene loss (CNA = -1). The model leverages
this data to compare mutation rates between normal and altered gene copy
states:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>gene_annotation <span class="ot">=</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, </span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="st">&quot;example_gene_annotation.csv.gz&quot;</span>, <span class="at">package =</span> <span class="st">&quot;mutmatch&quot;</span>))</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="fu">head</span>(gene_annotation)</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="co">#&gt;                          Sample    Gene   CNA</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="co">#&gt;                          &lt;char&gt;  &lt;char&gt; &lt;int&gt;</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="co">#&gt; 1: TCGA-EE-A2M5-06A-12D-A197-08  SAMD11    -1</span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a><span class="co">#&gt; 2: TCGA-EE-A2M5-06A-12D-A197-08   NOC2L    -1</span></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a><span class="co">#&gt; 3: TCGA-EE-A2M5-06A-12D-A197-08  KLHL17    -1</span></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a><span class="co">#&gt; 4: TCGA-EE-A2M5-06A-12D-A197-08 PLEKHN1    -1</span></span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a><span class="co">#&gt; 5: TCGA-EE-A2M5-06A-12D-A197-08    HES4    -1</span></span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a><span class="co">#&gt; 6: TCGA-EE-A2M5-06A-12D-A197-08   ISG15    -1</span></span></code></pre></div>
</div>
<div id="parameters-for-model-specification" class="section level3">
<h3>Parameters for Model Specification</h3>
<ul>
<li><code>neighborsWindow</code>: Defines the genomic window size around
the gene of interest, set to 0.5 Mb by default. This window specifies
the range for sampling neighboring genes to establish a background
mutation rate.</li>
<li><code>outlierNeighborsThreshold</code>: Controls the maximum allowed
variability in mutation rates between the gene of interest and its
neighbors. The default value of 0.2 permits up to 22% variability
(calculated as exp(0.2)-1).</li>
</ul>
</div>
</div>
<div id="adjusting-the-model-formula" class="section level2">
<h2>Adjusting the Model Formula</h2>
<p>You can modify the formula to include different variables based on
the study’s needs. For instance, adding a Cancer variable allows the
model to estimate selection pressures across various cancer types:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>selection_estimates_pancancer <span class="ot">&lt;-</span> <span class="fu">get_selection_estimates_neighbors</span>(</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>  <span class="at">hgnc =</span> <span class="st">&quot;KRAS&quot;</span>,</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>  <span class="at">mutationsPath =</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;example_mutations.csv.gz&quot;</span>, <span class="at">package =</span> <span class="st">&quot;mutmatch&quot;</span>),</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>  <span class="at">annotationGenePath =</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;example_gene_annotation.csv.gz&quot;</span>, <span class="at">package =</span> <span class="st">&quot;mutmatch&quot;</span>),</span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>  <span class="at">annotationGenomeWidePath =</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;example_genomewide_annotation.csv&quot;</span>, <span class="at">package =</span> <span class="st">&quot;mutmatch&quot;</span>),</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>  <span class="at">neighborsWindow =</span> <span class="st">&quot;0.5Mb&quot;</span>,</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>  <span class="at">outlierNeighborsThreshold =</span> <span class="fl">0.2</span>,</span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a>  <span class="at">formula =</span> <span class="st">&quot;MutationNumber ~ isTarget + CNA + isTarget:CNA + Cancer + Mutation + offset(log(ntAtRisk))&quot;</span>,</span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>  <span class="at">family =</span> <span class="st">&quot;bayes.poisson&quot;</span>,</span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>  <span class="at">simtimes =</span> <span class="dv">50</span></span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>) </span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a>selection_estimates_pancancer <span class="ot">=</span> <span class="fu">post_process_pvalues</span>(selection_estimates_pancancer)</span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a>selection_estimates_pancancer[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="fu">c</span>(<span class="st">&quot;coefName&quot;</span>, <span class="st">&quot;coef&quot;</span>, <span class="st">&quot;coef_corrected&quot;</span>, <span class="st">&quot;pVal&quot;</span>,<span class="st">&quot;pVal_corrected&quot;</span>)]</span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a><span class="co">#&gt;           coefName         coef coef_corrected         pVal pVal_corrected</span></span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a><span class="co">#&gt;             &lt;char&gt;        &lt;num&gt;          &lt;num&gt;        &lt;num&gt;          &lt;num&gt;</span></span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a><span class="co">#&gt; 1:       Intercept -14.45246500   -14.45246500 8.264699e-19             NA</span></span>
<span id="cb8-17"><a href="#cb8-17" tabindex="-1"></a><span class="co">#&gt; 2:       isTarget1   4.47370855     4.48536625 1.711890e-03              0</span></span>
<span id="cb8-18"><a href="#cb8-18" tabindex="-1"></a><span class="co">#&gt; 3:           CNA-1  -0.24775717    -0.24775717 7.749085e-01             NA</span></span>
<span id="cb8-19"><a href="#cb8-19" tabindex="-1"></a><span class="co">#&gt; 4: isTarget1:CNA-1  -0.23852167    -0.20190583 7.851223e-01              0</span></span>
<span id="cb8-20"><a href="#cb8-20" tabindex="-1"></a><span class="co">#&gt; 5:      CancerLUAD  -0.01051128    -0.01051128 9.853077e-01             NA</span></span></code></pre></div>
<p>Alternatively, by removing the CNA variable from the formula—while
still including the CNA annotations via the annotationGenePath—the model
can calculate the selection coefficients separately for each cancer type
and each level of CNA. These distinctions will then be displayed in the
Cohort column.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>selection_estimates_per_cna <span class="ot">&lt;-</span> <span class="fu">get_selection_estimates_neighbors</span>(</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>  <span class="at">hgnc =</span> <span class="st">&quot;KRAS&quot;</span>,</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>  <span class="at">mutationsPath =</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;example_mutations.csv.gz&quot;</span>, <span class="at">package =</span> <span class="st">&quot;mutmatch&quot;</span>),</span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>  <span class="at">annotationGenePath =</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;example_gene_annotation.csv.gz&quot;</span>, <span class="at">package =</span> <span class="st">&quot;mutmatch&quot;</span>),</span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>  <span class="at">annotationGenomeWidePath =</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;example_genomewide_annotation.csv&quot;</span>, <span class="at">package =</span> <span class="st">&quot;mutmatch&quot;</span>),</span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>  <span class="at">neighborsWindow =</span> <span class="st">&quot;0.5Mb&quot;</span>,</span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>  <span class="at">outlierNeighborsThreshold =</span> <span class="fl">0.2</span>,</span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a>  <span class="at">formula =</span> <span class="st">&quot;MutationNumber ~ isTarget + Mutation + offset(log(ntAtRisk))&quot;</span>,</span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a>  <span class="at">family =</span> <span class="st">&quot;bayes.poisson&quot;</span>,</span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a>  <span class="at">simtimes =</span> <span class="dv">50</span></span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a>) </span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a>selection_estimates_per_cna <span class="ot">=</span> <span class="fu">post_process_pvalues</span>(selection_estimates_per_cna)</span>
<span id="cb9-13"><a href="#cb9-13" tabindex="-1"></a>selection_estimates_per_cna[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="fu">c</span>(<span class="st">&quot;Cohort&quot;</span>, <span class="st">&quot;coefName&quot;</span>, <span class="st">&quot;coef&quot;</span>, <span class="st">&quot;coef_corrected&quot;</span>, <span class="st">&quot;pVal&quot;</span>,<span class="st">&quot;pVal_corrected&quot;</span>)]</span>
<span id="cb9-14"><a href="#cb9-14" tabindex="-1"></a><span class="co">#&gt; Key: &lt;Cohort&gt;</span></span>
<span id="cb9-15"><a href="#cb9-15" tabindex="-1"></a><span class="co">#&gt;       Cohort      coefName         coef coef_corrected         pVal</span></span>
<span id="cb9-16"><a href="#cb9-16" tabindex="-1"></a><span class="co">#&gt;       &lt;char&gt;        &lt;char&gt;        &lt;num&gt;          &lt;num&gt;        &lt;num&gt;</span></span>
<span id="cb9-17"><a href="#cb9-17" tabindex="-1"></a><span class="co">#&gt; 1: COREAD__0     Intercept -12.69255476   -12.69255476 3.117515e-17</span></span>
<span id="cb9-18"><a href="#cb9-18" tabindex="-1"></a><span class="co">#&gt; 2: COREAD__0     isTarget1   3.27819938     3.56423363 1.191858e-02</span></span>
<span id="cb9-19"><a href="#cb9-19" tabindex="-1"></a><span class="co">#&gt; 3: COREAD__0 MutationACA&gt;G  -0.01132096    -0.01132096 9.908600e-01</span></span>
<span id="cb9-20"><a href="#cb9-20" tabindex="-1"></a><span class="co">#&gt; 4: COREAD__0 MutationACA&gt;T  -0.01132096    -0.01132096 9.908600e-01</span></span>
<span id="cb9-21"><a href="#cb9-21" tabindex="-1"></a><span class="co">#&gt; 5: COREAD__0 MutationACC&gt;A   5.56017667     5.56017667 1.033908e-05</span></span>
<span id="cb9-22"><a href="#cb9-22" tabindex="-1"></a><span class="co">#&gt;    pVal_corrected</span></span>
<span id="cb9-23"><a href="#cb9-23" tabindex="-1"></a><span class="co">#&gt;             &lt;num&gt;</span></span>
<span id="cb9-24"><a href="#cb9-24" tabindex="-1"></a><span class="co">#&gt; 1:             NA</span></span>
<span id="cb9-25"><a href="#cb9-25" tabindex="-1"></a><span class="co">#&gt; 2:              0</span></span>
<span id="cb9-26"><a href="#cb9-26" tabindex="-1"></a><span class="co">#&gt; 3:             NA</span></span>
<span id="cb9-27"><a href="#cb9-27" tabindex="-1"></a><span class="co">#&gt; 4:             NA</span></span>
<span id="cb9-28"><a href="#cb9-28" tabindex="-1"></a><span class="co">#&gt; 5:             NA</span></span></code></pre></div>
<div id="considerations-for-different-distribution-models" class="section level3">
<h3>Considerations for Different Distribution Models</h3>
<p>If the dataset is sufficiently large (e.g., high mutation counts in
both the gene of interest and its neighbors), alternative statistical
distributions like “poisson” or “negative.binomial” may be used to model
mutation counts. However, smaller datasets may not provide reliable
estimates under high stratification:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>mutation_table <span class="ot">&lt;-</span> <span class="fu">get_neighbors_model_table</span>(</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>    <span class="at">hgnc =</span> <span class="st">&quot;KRAS&quot;</span>,</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>     <span class="at">mutationsPath =</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;example_mutations.csv.gz&quot;</span>, <span class="at">package =</span> <span class="st">&quot;mutmatch&quot;</span>),</span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>  <span class="at">annotationGenePath =</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;example_gene_annotation.csv.gz&quot;</span>, <span class="at">package =</span> <span class="st">&quot;mutmatch&quot;</span>),</span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>  <span class="at">annotationGenomeWidePath =</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;example_genomewide_annotation.csv&quot;</span>, <span class="at">package =</span> <span class="st">&quot;mutmatch&quot;</span>),</span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>     <span class="at">neighborsWindow =</span> <span class="st">&quot;0.5Mb&quot;</span>,</span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a>  <span class="at">outlierNeighborsThreshold =</span> <span class="fl">0.2</span>,</span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a>  )</span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a>mutation_table<span class="sc">$</span>MutationNumber <span class="sc">%&gt;%</span> <span class="fu">sum</span>() <span class="sc">%&gt;%</span> <span class="fu">print</span>()</span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a><span class="co">#&gt; [1] 7</span></span></code></pre></div>
</div>
</div>
<div id="conclusion" class="section level2">
<h2>Conclusion</h2>
<p>For more detailed parameter usage and additional functions, consult
the <em>mutmatch</em> documentation on specific functions, for example
<code>?get_selection_estimates_neighbors</code>. The flexibility of
<em>mutmatch</em> allows for extensive analysis tailored to specific
research needs.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
